source/AsyncRgbLedSimulationDataGenerator.h

```diff
@@ -19,11 +22,20 @@ class AsyncRgbLedSimulationDataGenerator
 	U32 mSimulationSampleRateHz;
 
 protected:
-	void CreateSerialByte();
-	std::string mSerialText;
-	U32 mStringIndex;
+	void CreateRGBWord();
+	RGBValue RandomRGBValue() const;
+
+    void WriteRGBTriple( const RGBValue& rgb );
+	void WriteUIntData( U32 data, U8 bit_count );
+	void WriteBit(bool b);
 
-	SimulationChannelDescriptor mSerialSimulationData;
+    void WriteReset();
+	
+    ClockGenerator mClockGenerator;
+	SimulationChannelDescriptor mLEDSimulationData;
 
+	// largest value for a color channel in the selected controller.
+	// this is 2^bitSize - 1
+	U32 mMaximumChannelValue = 255;
```

Comment:
since the value set here is never used (it's over-written in Initialize) should we still initialize it here?

Reply:
This is just force of habit for me at this point, in case someone adds new constructors or init code paths and forget to initialise members. But as you it can be removed if you prefer.